### PR TITLE
Fix issue with VisitorIdentification control on Next.js

### PR DESCRIPTION
**Pull request by ambrauer**
- Fixes an issue with `VisitorIdentification` React control on Next.js by moving everything (script and meta tag addition) to purely client-side 
- A pageEditing check has also been added within the control which avoids having to check in consuming app

### CODE REVIEW

File: packages/sitecore-jss-react/src/components/VisitorIdentification.tsx [hunk 1]
```diff
@@ -3,27 +3,37 @@ import { withSitecoreContext } from '../enhancers/withSitecoreContext';
...
 let emittedVI = false;
-let VIComponent: React.SFC<VIProps> = ({ sitecoreContext }) => {
+const VIComponent: React.FC<VIProps> = ({ sitecoreContext }) => {
   if (
-    !emittedVI &&
-    typeof document !== 'undefined' &&
-    sitecoreContext.visitorIdentificationTimestamp
+    emittedVI ||
+    typeof document === 'undefined' ||
+    sitecoreContext.pageEditing ||
```
Comment (sc-illiakovalenko) [inline packages/sitecore-jss-react/src/components/VisitorIdentification.tsx:16]: @ambrauer So, in Experience Editor this warning also exists?

Comment (sc-illiakovalenko) [reply packages/sitecore-jss-react/src/components/VisitorIdentification.tsx:16]: Nvm, I understood that it shouldn't be invoked in EE

File: packages/sitecore-jss-react/src/components/VisitorIdentification.tsx [hunk 2]
```diff
@@ -3,27 +3,37 @@ import { withSitecoreContext } from '../enhancers/withSitecoreContext';
...
+
+  const script = document.createElement('script');
+  script.src = `/layouts/system/VisitorIdentification.js`;
+  script.type = 'text/javascript';
+
+  const meta = document.createElement('meta');
+  meta.name = 'VIcurrentDateTime';
+  meta.content = sitecoreContext.visitorIdentificationTimestamp;
+
+  const head = document.getElementsByTagName('head')[0];
```
Comment (sc-illiakovalenko) [inline packages/sitecore-jss-react/src/components/VisitorIdentification.tsx:34]: @ambrauer What about `document.querySelector('head')`?

File: packages/sitecore-jss-react/src/components/VisitorIdentification.tsx (hunk 1 above)
Comment (sc-illiakovalenko) [reply packages/sitecore-jss-react/src/components/VisitorIdentification.tsx:16]: Need remember to add in release notes that pageEditing will be checked inside component

**Review by sc-illiakovalenko — CHANGES_REQUESTED** [review]

File: packages/sitecore-jss-react/src/components/VisitorIdentification.tsx (hunk 2 above)
Comment (ambrauer) [reply packages/sitecore-jss-react/src/components/VisitorIdentification.tsx]: @sc-illiakovalenko Probably safer, I've switched to use.

File: packages/sitecore-jss-react/src/components/VisitorIdentification.tsx (hunk 1 above)
Comment (ambrauer) [reply packages/sitecore-jss-react/src/components/VisitorIdentification.tsx]: @sc-illiakovalenko Actually removed sitecoreContext.pageEditing check after further testing. This is not necessary as sitecoreContext.visitorIdentificationTimestamp is always undefined in editing/preview mode. I believe when I noticed this issue it was during editing, but the main issue was adding the meta tag w/ VI timestamp at render time (which has been resolved).

File: samples/nextjs/src/components/Layout.tsx [hunk 3]
```diff
@@ -72,7 +66,7 @@ const Layout = ({ route, sitecoreContext }: LayoutProps) => {
 
         VI detection only runs once for a given analytics ID, so this is not a recurring operation once cookies are established.
       */}
-      {!sitecoreContext.pageEditing && <VisitorIdentification />}
```
Comment (sc-illiakovalenko) [inline samples/nextjs/src/components/Layout.tsx:75]: @ambrauer You removed page editing check inside VisitorIdentification component, should you revert changes here?

**Review by sc-illiakovalenko — CHANGES_REQUESTED** [review]

File: samples/nextjs/src/components/Layout.tsx (hunk 3 above)
Comment (ambrauer) [reply samples/nextjs/src/components/Layout.tsx:75]: @sc-illiakovalenko No, this check was made originally in the POC as a quick way to get past this issue with the VisitorIdentification control. But no longer required now that it's resolved.

**Review by sc-illiakovalenko — APPROVED** [review]

**Pull request merged**